File: scripts/supercommit.py
#!/usr/bin/env python3

import sys
import subprocess

if len(sys.argv) < 2:
    sys.exit("You need to provide a commit message")
msg = sys.argv[1]

subprocess.check_call("git submodule foreach git commit -m".split() + [msg])

subprocess.check_call("git add backend".split())
subprocess.check_call("git commit -m".split() + [msg])

File: scripts/aliases/supercommit.ts
import { commandSync, sync } from 'execa';
import { parse } from 'shell-quote';

if (process.argv.length < 2) {
	console.error('You need to provide a commit message.');
	process.exit(1);
}

const args = process.argv.slice(2);

try {
	sync(
		'git',
		[...parse('submodule foreach git commit'), ...args] as string[],
		{
			stdout: process.stdout,
		}
	);
} catch {
	// empty
}

commandSync('git add backend');
sync('git', ['commit', ...args]);
